### ISSUE TITLE
Tectonic seems to loop infinitely on this input file

### ISSUE THREAD

**Issue by pkgw** [tectonic-typesetting/tectonic#470]
I have *not* yet sat down and tried to create a minimal test case. xelatex processes the file OK.

Also, apparently GitHub filters file extensions, so let's call this a `.txt`.

[timeseries1.tex.txt](https://github.com/tectonic-typesetting/tectonic/files/3631476/timeseries1.tex.txt)

**Comment by ligfx** [comment]
I ran into the same issue with a Jupyter notebook exported to Latex, and minimized my file down to this:

```tex
\documentclass{article}

\usepackage{fancyvrb}
    
\makeatletter
\let\OriginalVerbatim=\Verbatim
\renewcommand{\Verbatim}[1][1]{
    \def\FV@Space{\discretionary{}{}{something} }
    \OriginalVerbatim[#1]}
\makeatother
    
\begin{document}

\begin{Verbatim}[]
hello world
\end{Verbatim}

\end{document}
```

**Comment by ralismark** [comment]
Running `gdb`, it appears that the tex engine gets stuck in [this loop](https://github.com/tectonic-typesetting/tectonic/blob/master/tectonic/xetex-xetex0.c#L17339-L17355). Neither of the ifs get run. Here are some relevant values, copied from gdb:

- `main_pp` = 525
- `main_ppp` = 525
- `is_char_node(main_ppp)` = false
- `mem[main_ppp]` = `{b32 = {s0 = 655360, s1 = -268435455}, b16 = {s0 = 0, s1 = 10, s2 = 1, s3 = 61440},
  gr = -3.1050391462465209e+231, ptr = 0xf0000001000a0000}`

I'm not familiar enough with the how the tex engine works to figure out what's causing this.

**Comment by pkgw** [comment]
Thanks for the investigation @ralismark ! The TeX engine should *really* not have any bugs of this nature, so this probably indicates a mistake in my translation of the core code. For these kinds of issues I usually end up consulting [the book](https://github.com/tectonic-typesetting/tectonic-staging/#generating-the-reference-book) and trying to trace through what's going on to see where I went wrong.

I hope to be able to take a deeper look at this myself, but my main priority is to get a TeXLive 2020 update rolled out.

**Comment by pkgw** [comment]
Fixed by #666 it seems!
